syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.c_login";
option java_outer_classname = "LoginProto";

package grpc_login;

message Result {
  int32 code = 1;
  string message = 2;
  User user = 3;
}

message User{
  int32 id = 1;
  string username = 2;
}
message SessionRequest{
  string token = 1;
}

message SessionResponse{
  int32 code = 1;
  string message = 2;
  string token = 3;
}

message UserRequest {
  string username = 1;
  string password = 2;
}

service LoginGrpcService {
  rpc registerUser  (UserRequest)    returns (Result) {}
  rpc login  (UserRequest)    returns (SessionResponse) {}
  rpc userInfo  (SessionRequest)    returns (Result) {}
  rpc logout  (SessionRequest)    returns (Result) {}
}

